components/automate-elasticsearch/habitat/hooks/post-run

```diff
@@ -0,0 +1,41 @@
+#!{{pkgPathFor "core/bash"}}/bin/bash
+
+exec 2>&1
+
+{{#if cfg.disable ~}}
+exit 0
+{{~else}}
+
+# ElasticSearch >= 6 doesn't allow configuring index settings in the elasticsearch.yml
+# so we need to configure these after ElasticSearch is up.
+
+# The post-run hook is fired immediately after run and doesn't wait until
+# elasticsearch is healthy. Therefore, we'll loop in the health-check until
+# it's green (0) or yellow (1) before we try and make the request.
+
+{{#if svc.me.cfg.http-host ~}}
+HOST="{{svc.me.sys.ip}}"
+{{else}}
+HOST="127.0.0.1"
+{{/if ~}}
+
+source {{pkg.svc_config_path}}/health_check
+
+if [[ $(wait_until_healthy 300) -ne 0 ]]; then
```

Comment:
i think past experience has shown blocking in hooks is not good. habitat-sh/habitat#6705 indicates that it will retry if you exit with non-zero. The ha services also did something similar to this but ended up ditching it because it caused some problems

Reply:
For posterity: previous versions of hab ran the hooks in the main supervisor thread so any long operation in them would block the supervisor. Now hooks are run async so it's okay to block.